ADRs/2022-12-08-PropertyWrappers-ADR.md

```diff
@@ -0,0 +1,184 @@
+## Problem Statement
+
+Using VSM with UIKit can be cumbersome. Specifically, when rendering state changes. In addition, this ADR explores the possibility of using property wrappers for a more holistic approach to working with the `StateContainer` type.
+
+Currently, UIKit views require developers to manually subscribe to state updates to update the view hierarchy when the state changes.
+
+Example:
+
+```swift
+import Combine
+
+class SomeViewController: UIViewController, ViewStateRendering {
+    var container: StateContainer<SomeViewState>
+    var stateSubscription: AnyCancellable?
+    ...
+    func viewDidLoad() {
+        stateSubscription = container
+            .$state
+            .sink { [weak self] newState in
+                self?.render(state: newState)
+            }
+    }
+
+    func render(state newState: SomeViewState) {
+        switch newState {
+            ...
+        }
+    }
+}
+```
+
+These are the problems with the current approach:
+
+1. Wiring up the subscription, including toting a subscription property, introduces repetitive boilerplate code to each VSM view.
```

Comment:
Hey @albertbori 👋 

**HEADS UP:** The numbers all start with 1.  I think you want 1-4.

Reply:
Hey @Greg-at-Wayfair, the markdown rendering standard automatically converts a list of `1.` items to a numbered html list, but if it makes it easier to review, I can manually set the numbers.